ibis/expr/types/arrays.py

```diff
@@ -825,7 +825,8 @@ class ArrayScalar(Scalar, ArrayValue):
 
 @public
 class ArrayColumn(Column, ArrayValue):
-    pass
+    def __getitem__(self, index: int | ir.IntegerValue | slice) -> ir.Column:
```

Comment:
Why to override this method? Only to change the return annotation?

Reply:
I guess we would need to turn Column to a generic type then which we can implement later.

Reply:
It's less invasive to not reorder inheritance. Fewer changes were needed. We can reorder the base classes later if we want to.